### PR TITLE
Convert Measure into an ExtendedEnum

### PR DESCRIPTION
**Pull request by cjkent**
Convert measure to an `ExtendedEnum` so it can contain state but still be referred to by name.

This is in preparation for tackling #740 (FX conversion flag on `Measure`) and #562 (add types to `Measure`).

The important changes are in:

    Measure
    Measures
    StandardMeasures
    ImmutableMeasure
    Measures.ini

The remaining changes are caused by moving the standard measure constants from `Measure` to `Measures`.

The standard measures are defined in code in `StandardMeasures`. Defining a text format for measures is beyond the scope of these changes. If we decide we need one it would be more sensible to think about it once we've made all the changes to `Measure`.

### CODE REVIEW

File: modules/calc/src/main/java/com/opengamma/strata/calc/config/Measure.java [hunk 1]
```diff
@@ -22,136 +22,30 @@
...
-  /**
-   * Obtains an instance from the specified name.
-   * <p>
-   * Measure names must only contains the characters A-Z, a-z, 0-9 and -.
+   * Obtains an instance from the specified unique name.
    *
-   * @param name  the name of the measure
-   * @return a measure with the specified name
+   * @param uniqueName  the unique name
+   * @return the convention
```
Comment (jodastephen) [inline modules/calc/src/main/java/com/opengamma/strata/calc/config/Measure.java]: Not a convention

File: modules/calc/src/main/java/com/opengamma/strata/calc/config/Measure.java [hunk 2]
```diff
@@ -22,136 +22,30 @@
...
+    return extendedEnum().lookup(uniqueName);
   }
 
   /**
-   * Creates an instance.
-   * 
-   * @param name  the name of the measure
+   * Gets the extended enum helper.
+   * <p>
+   * This helper allows instances of the convention to be looked up.
```
Comment (jodastephen) [inline modules/calc/src/main/java/com/opengamma/strata/calc/config/Measure.java]: Not a convention

File: modules/calc/src/test/java/com/opengamma/strata/calc/config/pricing/PricingRuleTest.java [hunk 3]
```diff
@@ -30,36 +31,32 @@
...
-  private static final Measure MEASURE1 = Measure.of("1");
-  private static final Measure MEASURE2 = Measure.of("2");
-  private static final Measure MEASURE3 = Measure.of("3");
-
   private static final FunctionGroup<TestTrade1> GROUP =
       DefaultFunctionGroup.builder(TestTrade1.class)
           .name("GroupName")
-          .addFunction(MEASURE1, FunctionConfig.of(TestFunction1.class))
-          .addFunction(MEASURE2, FunctionConfig.of(TestFunction2.class))
+          .addFunction(Measures.PRESENT_VALUE, FunctionConfig.of(TestFunction1.class))
```
Comment (jodastephen) [inline modules/calc/src/test/java/com/opengamma/strata/calc/config/pricing/PricingRuleTest.java]: Unfortunate that the static imports have been lost. Must also have left the file with an unused static import. Comment applies to numerous files.

Comment (cjkent) [inline modules/calc/src/test/java/com/opengamma/strata/calc/config/pricing/PricingRuleTest.java]: They haven't been lost, I've just replaced a locally defined constant with an externally defined constant.

Previously it was possible to define a local measure with an arbitrary name. Now they have to match the built in set of measures. In which case it makes sense to use the standard measures rather then defining local aliases.

**Pull request merged**